src/reader/reader_handler.py

```diff
@@ -29,26 +29,32 @@
 from numpy import random
 from time import sleep
 import glob
+from src.common.constants import MODULE_DATA_READER
+
+
+dlp = Profile(MODULE_DATA_READER)
 
 class FormatReader(ABC):
     read_images = None
-    def __init__(self, dataset_type, thread_index, epoch_number):
-        self.profile_args = {}
-        self.profile_args["epoch"] = epoch_number
+
+    def __init__(self, dataset_type, thread_index):
         self.thread_index = thread_index
         self._args = ConfigArguments.get_instance()
         logging.debug(
             f"Loading {self.__class__.__qualname__} reader on thread {self.thread_index} from rank {self._args.my_rank}")
         self.dataset_type = dataset_type
         self.open_file_map = {}
-        self.epoch_number = epoch_number
+
         if FormatReader.read_images is None:
             FormatReader.read_images = 0
-
+        self.image = None
```

Comment:
Now that we dont need to share image between get_sample and resize. Can u remove this and make it a local variable in the get_sample method?